contrib/corrosion-cmake/CMakeLists.txt

```diff
@@ -53,8 +53,9 @@ if(NOT ENABLE_RUST)
 endif()
 
 # Try to find proper rust toolchain one more time in case of ENABLE_RUST already installed to give sensible error
-find_package(Rust)
 if (DEFINED ENABLE_RUST)
+  list (APPEND CMAKE_MODULE_PATH "${ClickHouse_SOURCE_DIR}/contrib/corrosion/cmake")
```

Comment:
Shouldn't we remove this same thing in L66 since it's redundant?
Also, I think the if branch that checks if `ENABLE_RUST` is defined in L56 is not needed, since in L52 we return in case is not enabled already.

Reply:
>On L56 we check that the variable is defined - if (DEFINED ENABLE_RUST), it is different from if (NOT ENABLE_RUST) (on L52)

Right, but isn't `ENABLE_RUST` always defined to at least the `DEFAULT_ENABLE_RUST` as per L49 since `DEFAULT_ENABLE_RUST` is always defined in L35-L47?

Reply:
Oh, right, thanks.
Fixed it slightly better now.